lightwood/api/high_level.py

```diff
@@ -50,12 +50,24 @@ def json_ai_from_problem(df: pd.DataFrame, problem_definition: Union[ProblemDefi
     if not isinstance(problem_definition, ProblemDefinition):
         problem_definition = ProblemDefinition.from_dict(problem_definition)
 
+    started = time.time()
+
     log.info(f'Dropping features: {problem_definition.ignore_features}')
     df = df.drop(columns=problem_definition.ignore_features)
 
     type_information = infer_types(df, problem_definition.pct_invalid)
     stats = statistical_analysis(
         df, type_information.dtypes, type_information.identifiers, problem_definition)
+
+    duration = time.time() - started
+    if problem_definition.time_aim is not None:
+        problem_definition.time_aim -= duration
+        if problem_definition.time_aim < 10:
+            problem_definition.time_aim = 10
+
+    # Assume that the stuff besdies encoder and mixers takes about as long as analyzing did... bad, but let's see
```

Comment:
I think it would be important to add this into the `ProblemDefinition` docstring:

> expected_additional_time: stuff besides encoder and mixers

Also, typo: besides*

Reply:
done